src/trait_manager/model/Trait.ts

```diff
@@ -108,12 +108,21 @@ export default class Trait extends Model<TraitProperties> {
     const normPath = stringToPath(path || '').join('.');
     const dataListeners: DataVariableListener[] = [];
     const prevListeners = this.dataListeners || [];
+    const [ds, dr] = this.em.DataSources.fromPath(path);
 
     prevListeners.forEach((ls) => this.stopListening(ls.obj, ls.event, this.updateValueFromDataVariable));
 
+    ds && dataListeners.push({ obj: ds.records, event: 'add remove reset' });
+    dr && dataListeners.push({ obj: dr, event: 'change' });
     dataListeners.push({ obj: dataVar, event: 'change:value' });
     dataListeners.push({ obj: em, event: `${DataSourcesEvents.path}:${normPath}` });
 
+    dataListeners.push(
+      { obj: dataVar, event: 'change:path change:value' },
```

Comment:
```suggestion
      { obj: dataVar, event: 'change:path change:defaultValue' },
```

Reply:
I am unsure about this one, what is `value` referencing here?

Reply:
This is backbone's default events on property changes. As you changed from `value` to `defaultValue`, also the event has to be changed

Reply:
* https://github.com/GrapesJS/grapesjs/pull/6098#issuecomment-2322320530